tests/baselines/reference/user/puppeteer.log

```diff
@@ -30,6 +30,8 @@ lib/FrameManager.js(668,15): error TS2503: Cannot find namespace 'Protocol'.
 lib/JSHandle.js(32,15): error TS2503: Cannot find namespace 'Protocol'.
 lib/JSHandle.js(146,15): error TS2503: Cannot find namespace 'Protocol'.
 lib/JSHandle.js(237,29): error TS2503: Cannot find namespace 'Protocol'.
+lib/Launcher.js(179,17): error TS1003: Identifier expected.
```

Comment:
The parser for `@implements` must be wrong. This produces the error: `@implements {!Puppeteer.ProductLauncher}`, but this does not: `@implements {Puppeteer.ProductLauncher}`.

Reply:
Yes, it should be parsing a type reference, not an expression like `@extends` does. (Although perhaps `@extends` should not either?)

Reply:
Filed ##37095

Reply:
Why should `Puppeteer.ProductLauncher` be an error ? `ProductLauncher` is an interface so `Puppeteer.ProductLauncher` seems like something that can be implemented. 
When checking the implements expression there is an extra check that 

```ts
if (!isEntityNameExpression(typeRefNode.expression)) {
         error(typeRefNode.expression, Diagnostics.A_class_can_only_implement_an_identifier_Slashqualified_name_with_optional_type_arguments);
 }
```

So you can't have an arbitrary expression there, but you do get a more meaningful error message.